### ISSUE TITLE
UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape.

### ISSUE THREAD

**Issue by KihongHeo** [tflearn/tflearn#460]
Hi TFLearn Team,

I am new to tensorflow and tflearn and implementing a classifier using LSTM. 
In the meanwhileI, encountered the following warning:

/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gradients.py:90: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "

I cannot understand what it means. Could you give me a solution?

Here is a part of my code:

net = tflearn.input_data([None, 500]) 
net = tflearn.embedding(net, input_dim=len(dic), output_dim=128)
net = tflearn.embedding(net, input_dim=128, output_dim=128)
net = tflearn.lstm(net, 1024, dropout=0.8, dynamic=True)
net = tflearn.fully_connected(net, 2, activation='softmax')
net = tflearn.regression(net, optimizer='adam', learning_rate=args.rate,
                                loss='categorical_crossentropy')
model = tflearn.DNN(net, tensorboard_verbose=3)

Thanks.

**Comment by aymericdamien** [comment]
Maybe there is an issue with 2 embedding layer after each other.
Could you try to remove one to see if it is working?

**Comment by KihongHeo** [comment]
Sorry. It was my mistake during copy and paste. My actual code has only one embedding. (I fixed the above code)

**Comment by wight911** [comment]
I had the same warnings too when running another python code, but do not understand the means too.

**Comment by wight911** [comment]
just found a post on Stackoverflow. hope it may help. https://stackoverflow.com/questions/35892412/tensorflow-dense-gradient-explanation

**Comment by Nithanaroy** [comment]
Got this warning first time after moving from pure `keras` to `tf.keras`. Based on the above discussion in stackoverflow, looks like `tf.keras` code has to handle this.

**Comment by DrCMY** [comment]
Hi, is it possible to post the corrected version of the code?
Thank you
